processor/src/main/kotlin/dev/zacsweers/autoservice/ksp/AutoServiceSymbolProcessor.kt

```diff
@@ -82,8 +83,13 @@ public class AutoServiceSymbolProcessor(environment: SymbolProcessorEnvironment)
               return emptyList()
             }
 
-    resolver
-        .getSymbolsWithAnnotation(AUTO_SERVICE_NAME)
+    val symbols = resolver.getSymbolsWithAnnotation(AUTO_SERVICE_NAME)
+
+    val result = symbols.filter { !it.validate() }
+
+    symbols
+        .filterIsInstance<KSClassDeclaration>()
+        .filter { it.validate() }
         .filterIsInstance<KSClassDeclaration>()
```

Comment:
This double filterIsInstance also seems unnecessary. Let's try to make sure this remains a single pass iteration